mods/seasonal/moves.js

```diff
@@ -572,6 +572,61 @@ let BattleMovedex = {
 		target: "self",
 		type: "Normal",
 	},
+	// Quite Quiet
+	"spookytransform": {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "",
+		shortDesc: "",
+		id: "spookytransform",
+		name: "Spooky Transform",
+		pp: 5,
+		priority: 1,
+		flags: {mystery: 1},
+		onPrepareHit: function (target, source) {
+			this.attrLastMove('[still]');
+			return !source.removeVolatile('spookytransform');
+		},
+		onHit: function (target, pokemon) {
+			if (!pokemon.transformInto(target, pokemon)) {
+				return false;
+			}
+			this.add('-anim', pokemon, "Refresh", pokemon);
+			pokemon.addVolatile('spookytransform');
+		},
+		effect: {
+			onDamagePriority: -100,
+			onDamage: function (damage, target, source) {
+				if (!target.transformed) return damage;
+				this.add('-immune', target, '[msg]', '[from] effect: Spooky Transform');
+				this.add('-message', `${target.name} reflected the damage back to ${source.name}.`);
+				this.add('-anim', target, "Night Shade", source);
+				this.damage(damage, source);
+				return null;
+			},
+			onResidualOrder: 100,
+			onResidualSubOrder: 1,
+			onResidual: function (pokemon) {
+				pokemon.clearVolatile();
+				this.add('-clearboost', pokemon, false, '[silent]');
+				this.add('-end', pokemon, 'Transform');
+				pokemon.setAbility('Levitate');
+				pokemon.addVolatile('spookytransform');
```

Comment:
Why are you re-adding it here? Wouldn't that restart the effect? If your using this as a way of preventing the effect from being used twice in a row, you might want to check `pokemon.lastMove.id` instead.

Reply:
`clearVolatile` also clears `lastMove`. I did initially want to just copy most of clearVolatile here but it was a lot of code. Maybe it's the only way though...

Reply:
Also no it won't restart the effect because the volatile doesn't transform the Pokemon.

Reply:
You should try to avoid using clearVolatile if its clearing all of your volatiles. Perhaps try looping though your volatiles and clearing all but `quitequiet`